gradle-plugin-testing-junit/README.md

```diff
@@ -0,0 +1,606 @@
+
+```gradle
+plugins {
+    id 'com.palantir.idea-language-injector' version '<version>'
+}
+```
+
+This plugin automatically generates IntelliJ IDEA language injection configurations, enabling syntax highlighting for string literals.
+
+### Write your first test
+
+You can use `@BeforeEach` for shared setup:
+
+```java
+import com.palantir.gradle.testing.junit.GradlePluginTests;
+import com.palantir.gradle.testing.execution.GradleInvoker;
+import com.palantir.gradle.testing.project.RootProject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static com.palantir.gradle.testing.assertion.GradlePluginTestAssertions.assertThat;
+
+@GradlePluginTests
+class MyPluginTest {
+    @BeforeEach
+    void setup(RootProject project) {
+        project.buildGradle().plugins().add("my-plugin");
+        project.buildGradle().append("""
+                myPlugin {
+                    enabled = true
+                }
+                """);
+    }
+    
+    @Test
+    void plugin_applies_successfully(GradleInvoker gradle, RootProject project) {
+        assertThat(gradle.withArgs("tasks").buildsSuccessfully())
+            .output()
+            .contains("myTask");
+    }
+}
+```
+
+## Core Concepts
+
+### The `@GradlePluginTests` Annotation
+
+Mark test classes with `@GradlePluginTests` to enable Gradle plugin testing. This automatically:
+- Creates isolated project directories for each test
+- Runs tests against multiple Gradle versions
+- Provides parameter injection for test infrastructure
+
+### Parameter Injection
+
+Tests can request these parameters in any order:
+
+- **`GradleInvoker`** - Executes Gradle builds
+- **`RootProject`** - Root project directory and files
+- **`SubProject`** - Named subprojects (name derived from parameter name)
+
+Parameters work in `@Test`, `@BeforeEach`, and `@AfterEach` methods.
+
+**Example with subprojects:**
+```java
+@Test
+void multi_project_build(SubProject api, SubProject server) {
+    // Projects "api" and "server" are automatically created and included
+    api.buildGradle().plugins().add("java-library");
+
+    server.buildGradle().plugins().add("application");
+    server.buildGradle().append("""
+            dependencies {
+                implementation project(':api')
+            }
+            """);
+}
+```
+
+**Creating subprojects programmatically:**
+```java
+@Test
+void create_nested_subprojects_manually(SubProject api) {
+    // Nested subprojects
+    SubProject nestedProject = api.subproject("nested");
+}
+```
+
+### Multi-Version Testing
+
+The framework automatically runs each test against multiple Gradle versions.
+
+Configure which Gradle versions to test against using the `gradleTestUtils` extension:
+
+```gradle
+gradleTestUtils {
+    gradleVersions = ['8.10', '8.14.3']
+}
+```
+
+If not specified, tests run against the default version (currently `8.14.3`).
+
+### Testing with External Plugins
```

Comment:
maybe we should include here an example of how the test would fail if this setup is missing ?